calc/l3_route_resolver.go

```diff
@@ -63,7 +63,9 @@ type L3RouteResolver struct {
 	blockToRoutes          map[string]set.Set
 	allPools               map[string]model.IPPool
 	workloadIDToCIDRs      map[model.WorkloadEndpointKey][]cnet.IPNet
+	nodeToCIDRs            map[string]set.Set
```

Comment:
UT spotted that we weren't marking routes dirty when they targeted at Node and that node's IP changed.

e.g., the case where a WEP appears in the syncer before the corresponding node does, so we don't know the node's IP.

I added a new map to track the CIDRs for each node so that when the node IP changes we can mark those CIDRs dirty.

Reply:
That's meant to be handled by the call to `c.markAllNodeRoutesDirty(nodeName)`, but looks like that method is implemented with a loop over (all) blocks right now.

Adding a node to CIDR cache would make it more efficient but it'd need to be refcounted or something since you might have duplicate CIDRs on a node (at least transiently).